### ISSUE TITLE
Missing Version and TimeSent headers breaks wire compatibility with v6 for subscription messages

### ISSUE THREAD

**Issue by johnsimons** [Particular/NServiceBus#3583]
Here are the current headers being set:
```xml
<?xml version="1.0"?>
<ArrayOfHeaderInfo xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
	<HeaderInfo>
		<Key>NServiceBus.ControlMessage</Key>
		<Value>True</Value>
	</HeaderInfo>
	<HeaderInfo>
		<Key>NServiceBus.MessageIntent</Key>
		<Value>Subscribe</Value>
	</HeaderInfo>
	<HeaderInfo>
		<Key>SubscriptionMessageType</Key>
		<Value>Version_3_3.Messages.MyEvent, Version_3_3.Messages, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null</Value>
	</HeaderInfo>
	<HeaderInfo>
		<Key>NServiceBus.ReplyToAddress</Key>
		<Value>WireCompatVersion_6_0@JOHNPC2</Value>
	</HeaderInfo>
	<HeaderInfo>
		<Key>NServiceBus.SubscriberAddress</Key>
		<Value>WireCompatVersion_6_0@JOHNPC2</Value>
	</HeaderInfo>
	<HeaderInfo>
		<Key>NServiceBus.SubscriberEndpoint</Key>
		<Value>WireCompatVersion_6_0</Value>
	</HeaderInfo>
	<HeaderInfo>
		<Key>CorrId</Key>
		<Value />
	</HeaderInfo>
</ArrayOfHeaderInfo>
```

**Comment by johnsimons** [comment]
Not having a version header breaks v.3.3 compatibility, see https://github.com/Particular/NServiceBus/blob/02a7fdff135ced08b2263498244132d9c2f5dad2/src/unicast/NServiceBus.Unicast.BackwardCompatibility/IncomingSubscriptionMessageMutator.cs#L17-L26

**Comment by andreasohlund** [comment]
@WojcikMike pretty sure this happens since we now set those in the dispatch pipeline and subscriptions bypass that. We should make sure to use the dispatch pipeline in the subscription manager?

@johnsimons me an @WojcikMike can look at this

**Comment by johnsimons** [comment]
For subscriptions this is being addressed as part of #3576.
But I looked around and there seems other places that skip these headers.

**Comment by andreasohlund** [comment]
So there should be no direct usages of `IDispatchMessages` other than the dispatch terminator?

**Comment by johnsimons** [comment]
> So there should be no direct usages of IDispatchMessages other than the dispatch terminator?

I have no idea

**Comment by andreasohlund** [comment]
>So there should be no direct usages of IDispatchMessages other than the dispatch terminator?

My bad :)

Meant to say: There should be no direct usages of IDispatchMessages other than the dispatch terminator

**Comment by johnsimons** [comment]
ReadyMessageSender also uses IDispatchMessages directly

**Comment by danielmarbach** [comment]
I highlighted this in several discussions with @andreasohlund that I'm not
happy about our raw dispatch usages in the core. Unfortunately we failed to
come up with a good solution since then

**Comment by johnsimons** [comment]
I am pretty sure you guys can fix that later, after v6.0 has gone out :wink:

**Comment by andreasohlund** [comment]
>I am pretty sure you guys can fix that later, after v6.0 has gone out :wink:

Agreed, do we need to double check that the ready messages from v6 workers works with the v5 distributor? (we have no v6 distributor)

**Comment by danielmarbach** [comment]
Merged https://github.com/Particular/NServiceBus/pull/3576
